doc/guides/tfm/overview.rst

```diff
@@ -0,0 +1,266 @@
+
+Secure Processing Environment
+=============================
+
+Once the secure bootloader has finished executing, a TF-M based secure image
+will begin execution in the **secure processing environment**. This is where
+our device will be initially configured, and any secure services will be
+initialised.
+
+Note that the starting state of our device is controlled by the secure firmware,
+meaning that when the non-secure Zephyr application starts, peripherals may
+not be in the HW-default reset state. In case of doubts, be sure to consult
+the board support packages in TF-M, available in the ``platform/ext/target/``
+folder of TF-M (``modules/tee/tfm/trusted-firmware-m/``).
+
+Secure Services
+---------------
+
+As of TF-M 1.3.0, the following secure services are available:
+
+* Audit Logging (Audit)
+* Crypto (Crypto)
+* Firmware Update (FWU)
+* Initial Attestation (IAS)
+* Secure Storage, which has two parts:
+  * Internal Trusted Storage (ITS)
+  * Protected Storage (PS)
+
+A template also exists for creating your own custom services.
+
+For full details on these services, and their exposed APIs, please consult the
+`TF-M Documentation <https://tf-m-user-guide.trustedfirmware.org/>`__.
+
+Key Management and Derivation
+-----------------------------
+
+Key and secret management is a critical part of any secure device. You need to
+ensure that key material is available to regions that require it, but not to
+anything else, and that it is stored securely in a way that makes it difficult
+to tamper with or maliciously access.
+
+The **Internal Trusted Storage** service in TF-M is used by the **PSA Crypto**
+service (which itself makes use of mbedtls) to store keys, and ensure that
+private keys are only ever accessible to the secure processing environment.
+Crypto operations that make use of key material, such as when signing payloads
+or when decrypting sensitive data, all take place via key handles. At no point
+should the key material ever be exposed to the NS environment.
+
+One exception is that private keys can be provisioned into the secure
+processing environment as a one-way operation, such as during a factory
+provisioning process, but even this should be avoided where possible, and a
+request should be made to the SPE (via the PSA Crypto service) to generate a
+new private key itself, and the public key for that can be requested during
+provisioning and logged in the factory. This ensures the private key
+material is never exposed, or even known during the provisioning phase.
+
+TF-M also makes extensive use of the **Hardware Unique Key (HUK)**, which
+every TF-M device must provide. This device-unique key is used by the
+**Protected Storage** service, for example, to encrypt information stored in
+external memory, such that removing the memory and placing it on a new device,
+the contents of flash memory can't be decrypted since each device has it's own
+unique HUK used while encrypted the memory contents the first time.
```

Comment:
```suggestion
external memory. For example, this ensures that the contents of flash memory
can't be decrypted if they are removed and placed on a new device, since each
device has its own unique HUK used while encrypting the memory contents
the first time.
```

I had a hard time parsing this the first time I read it. Maybe something like this?